hazelcast/include/hazelcast/util/BlockingConcurrentQueue.h

```diff
@@ -37,7 +37,7 @@ namespace hazelcast {
             BlockingConcurrentQueue(size_t max_queue_capacity) : capacity_(max_queue_capacity), is_interrupted_(false) {
             }
 
-            void push(const T &e) {
+            void push(const T e) {
```

Comment:
```suggestion
            void push(T e) {
```
If this is left as const, call to std::move below won't cause the move ctor to be called, it will be a regular copy.

Reply:
good catch.